### DISCUSSION TITLE
Does `server.js` have a runtime dependency on `@remix-run/dev`?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [remix-run/remix#5420] [Q&A] [answered]
Here is a typical `server.js` for Cloudflare Pages:

```js
import { createPagesFunctionHandler } from "@remix-run/cloudflare-pages";
import * as build from "@remix-run/dev/server-build";

export const onRequest = createPagesFunctionHandler({
  build,
  getLoadContext: (context) => context,
  mode: process.env["NODE_ENV"] ?? "production",
});
```

What's the deal with line 2? Why am I importing something from the Remix CLI package to pass to my server adapter?

Does this mean there's currently a runtime dependency on the Remix CLI package in every Cloudflare Pages site? If so, maybe this could use some cleaning up.

**Comment by mjackson** [comment, marked as answer]
No, that isn't a runtime dependency. That is a virtual module id that is resolved at build time when the server is built. The purpose is to enable single-file builds of your server so our compiler can produce something suitable for running directly on CF workers.

**Reply by aaronadamsCA** [reply]
Got it, thank you! I was getting a type mismatch there during an attempted Remix upgrade, which led me down the rabbit hole of "what is this?" Appreciate the explanation.
